frontend/pages/policies/PolicyPage/components/PolicyForm/PolicyForm.tsx

```diff
@@ -178,6 +225,30 @@ const PolicyForm = ({
     !policyIdForEdit &&
     DEFAULT_POLICIES.find((p) => p.name === lastEditedQueryName);
 
+  useEffect(() => {
+    setSelectedTargetType(
+      !lastEditedQueryLabelsIncludeAny.length &&
+        !lastEditedQueryLabelsExcludeAny.length
+        ? "All hosts"
+        : "Custom"
+    );
+    setSelectedCustomTarget(
+      lastEditedQueryLabelsExcludeAny.length
+        ? "labelsExcludeAny"
+        : "labelsIncludeAny"
+    );
+    setSelectedLabels(
+      lastEditedQueryLabelsIncludeAny
+        .concat(lastEditedQueryLabelsExcludeAny)
+        .reduce((acc, label) => {
+          return {
+            ...acc,
+            [label.name]: true,
+          };
+        }, {}) || {}
+    );
+  }, [lastEditedQueryLabelsIncludeAny, lastEditedQueryLabelsExcludeAny]);
+
```

Comment:
This useEffect is needed b/c the initial values come from the policy context, but that itself isn't populated until the API request resolves, so the _initial_ value set in `useState()` is empty for all of them.